### ISSUE TITLE
[REQUEST] Add the ability to include directly Pytorch Leaning Rate schedulers

### ISSUE THREAD

**Issue by jamartinh** [takuseno/d3rlpy#13]
**Describe the solution you'd like**
Add the ability to include Learning Rate schedulers directly fomr Pytorch already implemented.

[https://pytorch.org/docs/stable/optim.html](https://pytorch.org/docs/stable/optim.html)

**Comment by takuseno** [comment]
@jamartinh Hello. Thanks for your proposal!
Currently, the learning rate scheduler is not supported by default. But, recently, I've added `OptimizerFactory` class that gives you more flexible control over optimizers.
https://d3rlpy.readthedocs.io/en/latest/references/optimizers.html

I'm thinking about letting this have learning rate scheduler option.

**Comment by jamartinh** [comment]
@takuseno With respect this, I am thinking on an alternative to avoid the use of Callbacks.
I think Callbacks should be not the solution to this but insead an overhead and a real unnecessary pain to define such a callbacks API. I will propose you instead to include for every fit and fit_online method the additionals.

fitter and fitter_online methods. This methods do the same exact thing however instead of continuing iterating the "yield" so the become generators.
Hence at each iteration it will be easy to monitor, control, debug and change and share parameters of the Optimizers, Networks, Algorithms and so on, indeed t then will be easy to run iterations on different instances of the env, share buffers between algo instances, do some offline learning and the continue iterating over the fitter and fitter_online generators.

I am working on that and have an initial working version, to basic but its easy.

**Comment by takuseno** [comment]
Currently, it's not planned to support learning rate schdulers. But, we can do it like this.
https://github.com/takuseno/d3rlpy/blob/master/reproductions/offline/iql.py

**Comment by takuseno** [comment]
Finally, the learning rate scheduler has been implemented officially. Please check this example: https://github.com/takuseno/d3rlpy/blob/master/examples/lr_scheduler.py .

**Comment by jamartinh** [comment]
Good 👍
